src/gui/qgsmaptoolcapture.cpp

```diff
@@ -399,7 +399,20 @@ int QgsMapToolCapture::fetchLayerPoint( const QgsPointLocator::Match &match, Qgs
       QgsVertexId vId;
       if ( !f.geometry().vertexIdFromVertexNr( match.vertexIndex(), vId ) )
         return 2;
+
       layerPoint = f.geometry().constGet()->vertexAt( vId );
+
+      // ZM support depends on the target layer
+      if ( layerPoint.is3D() && !QgsWkbTypes::hasZ( vlayer->wkbType() ) )
```

Comment:
Silly micro-optimisation, but all the dropZValue/dropMValue implementations already have a check in them to make sure the geometry is3D/isMeasure before proceeding - so they are safe to call without this extra duplicate check if you want raw SPEEEEEEED! ;)